cmd/snap-exec/main.go

```diff
@@ -0,0 +1,109 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2014-2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"syscall"
+
+	"github.com/jessevdk/go-flags"
+
+	"github.com/snapcore/snapd/snap"
+)
+
+// for the tests
+var syscallExec = syscall.Exec
+
+func main() {
+	if err := run(); err != nil {
+		fmt.Printf("cannot snap-exec: %s\n", err)
+		os.Exit(1)
+	}
+}
+
+func run() error {
+	var opts struct {
+		Positional struct {
+			SnapApp string `positional-arg-name:"<snapApp>" description:"the application to run, e.g. hello-world.env"`
+		} `positional-args:"yes" required:"yes"`
+
+		Command string `long:"command" description:"use a different command like {stop,post-stop} from the app"`
+	}
+
+	parser := flags.NewParser(&opts, flags.HelpFlag|flags.PassDoubleDash)
+	args, err := parser.Parse()
+	if err != nil {
+		return err
+	}
+
+	// the SNAP_REVISION is set by `snap run`
+	revision := os.Getenv("SNAP_REVISION")
```

Comment:
We need to figure out the current revision so that we can load the right `meta/snap.yaml` (below in snap.ReadInfo(). The SNAP_REVISION environment is set by `snap run` as part of the general snap environment setup (it also sets up `SNAP`, `SNAP_DATA` etc), see #1247 . 

Having said this, we may not need the revision, we could teach snap.ReadInfo() to support "current" somehow. Is that the suggestion you have or am I misunderstanding?